doc/adrs/124_add_support_for_libraries.adoc

```diff
@@ -0,0 +1,37 @@
+= ADR-124 - Add support for libraries
+
+== Context
+
+Users want to store and have access to read only data (a.k.a libraries).
+
+== Decision
+
+=== Backend
+
+Introducing a new `IEditingContextProcessor` allowing to execute operations on editing context before and after its loading in `EditingContextSearchService`.
+
+```
+public interface IEditingContextProcessor {
+    void preProcess(IEditingContext editingContext);
+    void postProcess(IEditingContext editingContext);
+}
```

Comment:
Its not clear to me from the ADR when these are invoked wrt to the editing context's lifecycle.

Reply:
The sentence above says: "Introducing a new `IEditingContextProcessor` allowing to execute operations on editing context before and after its loading in `EditingContextSearchService`."
So `preProcess` and `postPrcoess` are invoked in `EditingContextSearchService`.
I can make it more explicit if you want.

Reply:
Sorry, my bad, I read to quickly.